app/services/ListRetrievalService.scala

```diff
@@ -32,8 +32,30 @@ class ListRetrievalService @Inject() (listRepository: ListRepository)(implicit e
   private def getVersion = MetaData("version", LocalDate.of(2020, 11, 5))
 
   def getList(listName: ListName): Future[Option[ReferenceDataList]] =
-    listRepository.getList(listName, getVersion).map {
-      x => Some(ReferenceDataList(listName, getVersion, x))
+    listRepository.getListByName(listName, getVersion).map {
+      x =>
+        Some(ReferenceDataList(listName, getVersion, x))
```

Comment:
Should we do a check here to ensure that the list is non-empty otherwise return None?